packages/connectivity/src/scp-cf/destination/destination-service.ts

```diff
@@ -116,9 +116,16 @@ async function fetchDestinations(
 
   return callDestinationEndpoint(targetUri, headers, options)
     .then(response => {
-      const destinations: Destination[] = response.data.map(d =>
-        parseDestination(d)
-      );
+      const destinations: Destination[] = response.data.reduce((all, d) => {
```

Comment:
nitpick:

I think this would be more readable in such a form:

```
      const destinations: Destination[] = response.data.filter(d => canParseDestination(d)).map(d =>
        parseDestination(d)
      );
```

Where `canParseDestination` is a new function that performs the required validations but does not throw an error/exception in the bad case but returns a boolean.

Is there a reason why this would be a bad idea?